gocourse05/camera/infrared.go

```diff
@@ -0,0 +1,60 @@
+package camera
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/google/uuid"
+)
+
+func NewInfraredCameraData(id int, animal, movement string) Data {
+	return Data{
+		ID:       id,
+		Animal:   animal + "_" + uuid.NewString(),
+		Movement: movement,
+	}
+}
+
+type InfraredCamera struct {
+	ID        int
+	Name      string
+	Data      []Data
+	lightType int
+}
+
+func NewInfraredCamera(id int, name string, data []Data) InfraredCamera {
+	return InfraredCamera{
+		ID:        id,
+		Name:      name,
+		Data:      data,
+		lightType: infraredLight,
+	}
+}
+
+func (ic InfraredCamera) RetrieveData() ([]Data, error) {
+	if len(ic.Data) == 0 {
+		return nil, errors.New("no camera data found")
+	}
+
+	return ic.Data, nil
+}
+
+func (ic InfraredCamera) ProcessData() (*ProcessedData, error) {
```

Comment:
Здається, так можна спростити код:
```suggestion
func (ic InfraredCamera) ProcessData() (ProcessedData, error) {
```